samples/simulation/ising/simple/SimpleIsing.qs

```diff
@@ -141,4 +153,52 @@ namespace Microsoft.Quantum.Samples.SimpleIsing {
     internal function AddMagnetization(current : Double, spinMeasurement : Result) : Double {
         return current + (spinMeasurement == One ? 0.5 | -0.5); 
     }
+
+    /// # Summary
+    /// Create an array of site-to-site couplings.
+    /// This supports either a chain, cycle or all-to-all coupled Ising lattice.
+    /// Here we use a constant Ising coupling J. Optionally the value can be 
+    /// set to -J, depending on the parity of the lattice index.
+    ///
+    /// # Input
+    /// ## nSites
+    /// Number of sites in Ising lattice
+    /// ## J
+    /// Site-to-site coupling parameter
+    /// ## signByParity
+    /// Set sign of coupling parameter by the index parity
+    /// ## cycle
+    /// Couple the last lattice site in the chain to the first
+    /// ## allToAll
+    /// Couple all lattice sites to each other
+    ///
+    /// # Result
+    /// ## couplings
+    /// Array of arrays with coupling parameters
+    function GenerateCouplings(
+        nSites : Int,
+        J : Double,
+        signByParity: Bool,
+        cycle: Bool,
+        allToAll: Bool
+    ) : (Int, Double)[][] {
+        mutable couplings = new (Int, Double)[][nSites];
```

Comment:
I don't quite see why this is an array of arrays. From the code it seems that each entry is an array with one element.

Reply:
From the [Q# Array types docs](https://docs.microsoft.com/en-us/quantum/user-guide/language/types#array-types) I read that `(Int, Double)[][]` should be an array of arrays, so the inner array is an array of `(Int, Double)` tuples.
Based on this comment I do realize I have implemented `allToAll` incorrectly because right now it's replacing `couplings[i]` with another array of length 1.